### ISSUE TITLE
Create OwnerPlan observable site-wide

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#442]

**Comment by situatedbit** [comment]
I have some other ideas how this might be implemented with some more flexibility to change in the future, but it seems the goal of this implementation is to be able to disable access features. Does it actually involve implementing the restricted access, or is that a separate story?

**Comment by situatedbit** [comment]
Alternate idea: 
Since these plans may shift, I recommend against using a numeric mechanism for determining if a payment plan allows for a particular feature. Instead, we should make these declarative. 

(I'm loosely following the patterns used in the [CanCan ruby gem](https://github.com/ryanb/cancan)).

I suggest we create ``Ability`` objects that can be asked, _can the user do X?_. Each plan level will have it's own ``Ability`` object. The ``Ability`` object can be queried like this:

```javascript
var ability = new OwnerProAbility(); // an instance of Ability

ability.canCreateMultipleListings(); // => true/false
```

The user model can have an ``Ability`` object based on the payment plan for that user. Then we just query the user object rather than worrying about a separate model. With a method like this, an activity like service professional services doesn't need to know about details pertaining to payment plans. It just needs to know if a user can do this or that.

When we create new plans, then all we have to do is create a new implementation of an ability object, and none of the other code changes.

**Comment by situatedbit** [comment]
(Under the method I'm describing above, we can still store payment plan statuses as numbers, but we keep those numbers internal to the payment plan model. They behave more like ID's than capability levels.)

**Comment by situatedbit** [comment]
I see. #328 depends on this. This is just infrastructure?

**Comment by situatedbit** [comment]
@IagoSRL Before we can estimate this item, what do we need to decide about it?

**Comment by IagoSRL** [comment]
I like the idea of `Ability`.

I agree too with add an ability property to the User model, that will be computed against the user payment plan; we need to add a property for the plan to both Model and API (`LcRest.UserProfile` class at C#); in the server, it exist a property for the plan at the `LcRest.UserPaymentPlan.paymentPlan` (through the API /me/payment-plan/), so we can add a copy of that property to the result of UserProfile (read-only).

Note that in the server we have a column at `users` table `ownerStatusID` (available at UserProfile class), but is not in use still (something added previous any implementation it seems); the current status for the payment plan is at `UserPaymentPlan.PlanStatus` but is a string given by Braintree; @joshdanielson Must this statusID be set based on the Braintree PlanStatus (there is a `OwnerStatus` table, values there are the Braintree values?).

It's important for the internal implementation of the Ability, I suppose it must check both the owner payment plan and the owner status in order to set-up the correct rights.

**Comment by joshdanielson** [comment]
@iagosrl I think the only clarification that may be missing is that according to our bylaws, a service professional isn't an official owner just because they have a payment plan with braintree. They have to complete the owner acknowledgment, listing, and two bookings. But for simplicity purposes, I think I'm willing to bend on this or do it manually to avoid complexity and just treat anyone paying the user fees as an owner for now. One workaround for this could be to do some sort of check when they do the owner acknowledgment that they've completed at least one listing and paid their user fees. That way we can use the acknowledgment date as their official owner date. Does that make sense? Help me out with reducing complexity please :)

**Comment by IagoSRL** [comment]
Sorry, forgot that details. So, that's the difference with the `OwnerStatusID` field at `users` table with the status given by Braintree that we store as `UserPaymentPlan.PlanStatus`.

Then, the 'ability' must check the `OwnerStatusID` (that need to be added to the model, already returned by the UserProfile API) to know if enable the features for the plan associated to the user.

Out of the scope of this issue, we need to implement the server procedure that checks all our bylaws rules to set the owner status. That's the issue #279 (I forgot this, is at Phase 2). So is clear now for me.

**Comment by situatedbit** [comment]
Two tasks:
- [ ] modify API
- [ ] create ability object/access point within app
